resources/resource_utils.py

```diff
@@ -0,0 +1,60 @@
+from typing import Optional, Dict
+import hail as hl
+
+
+# Resource classes
+class _Resource:
+    """
+    Generic abstract resource class.
+    """
+    def __init__(self, path: str, source_path: Optional[str] = None, **kwargs):
+        if type(self) is _Resource:
+            raise TypeError("Can't instantiate abstract class _Resource")
+
+        self.source_path = source_path
+        self.path = path
+        self.__dict__.update(kwargs)
```

Comment:
Since `source_path` is already optional and unused in some resources and the arbitrary keyword args seem to only be used for importing data (`dbsnp.vcf_header_path`), maybe those could be combined into one optional dict argument `import_sources`. That would isolate use of arbitrary attributes to only reference_data_import.py, which already treats each individual resource differently.

Reply:
So, the big advantage with classes is the ability to have TableResource and TableMatrixResource that each can load their corresponding resource automatically. I think it's so much nicer this way.

Now, I agree that using kwargs may not be the best option here. Maybe just an optional attributes Dict would be better? We need it for the occasional additional file that should be bundled but it'll be a minor use case.

Reply:
As for using a dict for `import_sources`, if others find this better I'm fine with it but I don't really see the advantage over having a clear optional `source_path` attribute.